### PR TITLE
refactor: deploy the training-operator with kubernetes resources

### PR DESCRIPTION
**Pull request by DnPlas**
This commit refactors the way the training-operator is deployed, as instead of using a sidecar container that runs the workload, we are now applying the Deployment and all the Kubernetes resources required by the training-operator controller to be able to mange training resources.
We are introducing this change in preparation for the upcoming 1.8 version, as it introduces the hard dependency on a Kubernetes Secret mounted in a volume for the training-operator workload to start. For more details please refer to canonical/training-operator#159.

### CODE REVIEW

**Review by orfeas-k — COMMENTED** [review]
left some minor comments and I 'm testing now

File: src/charm.py
Comment (DnPlas) [reply src/charm.py:53]: I guess you mean the 

```python
self._k8s_resource_handler = None
self._crd_resource_handler = None
```
This is just left as None to allow us to create a handler with [this](https://github.com/canonical/training-operator/blob/main/src/charm.py#L86) property.

btw, they are in [main](https://github.com/canonical/training-operator/blob/main/src/charm.py#L66C1-L67C42) as well, this PR just re-order them.

File: src/charm.py
Comment (DnPlas) [reply src/charm.py:119]: Mmmh, I think this is a typo, not sure how this end up here.

Comment (DnPlas) [reply src/charm.py:119]: I have removed it.

File: src/templates/rbac_manifests.yaml.j2
Comment (DnPlas) [reply src/templates/rbac_manifests.yaml.j2:11]: Updated.

**Review by orfeas-k — APPROVED** [review]

**Comment by DnPlas** [comment]
@orfeas-k COS integration tests were failing, this PR may need a second review.

**Review by orfeas-k — APPROVED** [review]

**Pull request merged**
